src/coprocessor/dag/expr/arithmetic.rs

```diff
@@ -197,11 +197,94 @@ impl FnCall {
             None => Ok(None),
         }
     }
+
+    pub fn intdivide_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let lhs = try_opt!(self.children[0].eval_int(ctx, row));
+        let rhs = try_opt!(self.children[1].eval_int(ctx, row));
+
+        if rhs == 0 {
+            return Ok(None);
+        }
+
+        let lus = mysql::has_unsigned_flag(self.children[0].get_tp().get_flag());
+        let rus = mysql::has_unsigned_flag(self.children[1].get_tp().get_flag());
+
+        let res = match (lus, rus) {
+            (true, true) => Ok(((lhs as u64) / (rhs as u64)) as i64),
+            (false, false) => div_i64(lhs, rhs),
+            (false, true) => div_i64_with_u64(lhs, rhs as u64).map(|r| r as i64),
+            (true, false) => div_u64_with_i64(lhs as u64, rhs).map(|r| r as i64),
+        };
+        res.map(Some)
+    }
+
+    pub fn intdivide_decimal(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let lhs = try_opt!(self.children[0].eval_decimal(ctx, row));
+        let rhs = try_opt!(self.children[1].eval_decimal(ctx, row));
+        let overflow = Error::overflow("BIGINT", &format!("({} DIV {})", lhs, rhs));
+        match lhs.into_owned() / rhs.into_owned() {
```

Comment:
It seems we could use the function `divide_decimal` and convert the response to i64?

Reply:
I'm afraid we can't: divide_decimal and int_divide_decimal has different overflow error message('DECIMAL value is out of range in (xx / xx)' vs 'BIGINT value is out of range in (xx div xx)'. Future more,  we cannot convert this message through some regular expression, there is no way we can judge if an overflow error is returned by the division of divide_decimal(), or by the evaluation of the children(lhs/rhs)